servers/zms/src/main/java/com/yahoo/athenz/zms/NotificationManager.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2019 Oath Holdings Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.athenz.zms;
+
+import com.yahoo.athenz.common.server.notification.Notification;
+import com.yahoo.athenz.common.server.notification.NotificationService;
+import com.yahoo.athenz.common.server.notification.NotificationServiceFactory;
+import com.yahoo.athenz.zms.store.AthenzDomain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.yahoo.athenz.common.server.notification.NotificationService.NOTIFICATION_TYPE_MEMBERSHIP_APPROVAL;
+
+public class NotificationManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationManager.class);
+
+
+    private NotificationServiceFactory notificationServiceFactory;
+    private NotificationService notificationService;
+    private ScheduledExecutorService scheduledExecutor;
+    private final DBService dbService;
+    private String userDomain;
```

Comment:
why are we keeping the userDomain since we don't seem to use it - we only use the userDomainPrefix.

and why are we not passing the userDomainPrefix as part of the constructor since ZMSImpl already has that value?